velox/functions/prestosql/Probability.h

```diff
@@ -61,5 +62,20 @@ struct NormalCDFFunction {
   }
 };
 
+template <typename T>
+struct PoissonCDFFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(double& result, double lambda, int64_t value) {
+    constexpr double kInf = std::numeric_limits<double>::infinity();
+
+    VELOX_USER_CHECK_GE(value, 0, "value must be a non-negative integer");
+    VELOX_USER_CHECK_NE(lambda, kInf, "lambda doesn't accept infinity values");
```

Comment:
This isn't an error in the Presto java function. We should not have this.

What happens if this function is passed kInf ? If it throws then we should catch the exception in the test and assert it there.

Reply:
@aditi-pandit It throws and error msg saying 
"Error in function boost::math::poisson_distribution<double>::poisson_distribution: Mean argument is inf, but must be > 0 !"

Reply:
Its better not to have this error message "lambda doesn't accept infinity values" in the function and catch exception in the caller. The "Continued fraction diverged to NaN" could be coming from internal java library used by the Presto server. 

Please update this code with that behavior.